C/typeInference.c

```diff
@@ -310,7 +311,7 @@ static bool typeInference(unification_arrow* arrow, const dag_node* dag, const s
         }          } };
       APPLY_BINDING(&(arrow[dag[i].child[0]].source), &((binding)
         { .kind = PRODUCT
-        , .arg = { &(arrow[i].source), word256Type }
+        , .arg = { &(arrow[i].source), &(bound_var[9]) }
```

Comment:
Can't we replace this `9` with `word256` (I guess, we'd need to promote `enum TypeNamesForJets` from primitives.c to primitives.h.)

It then seems to me that the dozen+ comments referring to 9 by value can go away.

Reply:
Done.